hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/OpcodeServiceTest.java

```diff
@@ -0,0 +1,415 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.web3.service;
+
+import static com.hedera.mirror.common.util.CommonUtils.instant;
+import static com.hedera.services.stream.proto.ContractAction.ResultDataCase.OUTPUT;
+import static com.hedera.services.stream.proto.ContractAction.ResultDataCase.REVERT_REASON;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.Mockito.doAnswer;
+
+import com.hedera.mirror.common.domain.entity.Entity;
+import com.hedera.mirror.common.domain.entity.EntityId;
+import com.hedera.mirror.common.domain.entity.EntityType;
+import com.hedera.mirror.common.domain.token.TokenPauseStatusEnum;
+import com.hedera.mirror.common.domain.transaction.EthereumTransaction;
+import com.hedera.mirror.common.domain.transaction.TransactionType;
+import com.hedera.mirror.rest.model.OpcodesResponse;
+import com.hedera.mirror.web3.common.TransactionHashParameter;
+import com.hedera.mirror.web3.common.TransactionIdOrHashParameter;
+import com.hedera.mirror.web3.common.TransactionIdParameter;
+import com.hedera.mirror.web3.evm.contracts.execution.OpcodesProcessingResult;
+import com.hedera.mirror.web3.evm.contracts.execution.traceability.OpcodeTracerOptions;
+import com.hedera.mirror.web3.evm.store.accessor.EntityDatabaseAccessor;
+import com.hedera.mirror.web3.exception.EntityNotFoundException;
+import com.hedera.mirror.web3.service.model.ContractDebugParameters;
+import com.hedera.mirror.web3.utils.ContractFunctionProviderEnum;
+import com.hedera.mirror.web3.utils.ResultCaptor;
+import com.hedera.mirror.web3.utils.TransactionProviderEnum;
+import com.hedera.node.app.service.evm.store.models.HederaEvmAccount;
+import java.math.BigInteger;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.SneakyThrows;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.springframework.beans.factory.annotation.Autowired;
+
+class OpcodeServiceTest extends ContractCallTestSetup {
```

Comment:
```suggestion
@RequiredArgsConstructor
class OpcodeServiceTest extends ContractCallTestSetup {
```

and

- remove `@Autowired` instance properties
- make them `final`